[hwe] Revisit driver list for HWE kernel

rucoder opened the pull request:
# Update built-in driver list to support initrd-less boot

Booting without an initrd means the kernel cannot rely on early userspace to autoload modules
(udev/mdev/nlplug). Make storage and host drivers that can host the root filesystem built-in (=y)
so the kernel can discover and mount root directly across common platforms.

### x86/PCI
- CONFIG_VMD=y              # Intel VMD so NVMe behind VMD is visible early

### Block / Virt
- CONFIG_VIRTIO_BLK=y       # root on virtio disks without initrd

### NVMe
- CONFIG_BLK_DEV_NVME=y     # NVMe root without module autoloading

### SCSI / RAID HBAs (server boot devices)
- CONFIG_SCSI_HPSA=y
- CONFIG_MEGARAID_SAS=y
- CONFIG_SCSI_MPT3SAS=y     # common enterprise boot HBAs built-in

### USB host & storage (USB stick / USB CD/DVD)
- CONFIG_USB_XHCI_HCD=y
- CONFIG_USB_XHCI_PCI_RENESAS=y
- CONFIG_USB_XHCI_PLATFORM=y
- CONFIG_USB_EHCI_HCD=y
- CONFIG_USB_STORAGE=y
- CONFIG_USB_UAS=y          # UAS when available

### MMC/SD (eMMC/SD cards)
- CONFIG_MMC_SDHCI=y
- CONFIG_MMC_SDHCI_PCI=y
- CONFIG_MMC_SDHCI_ACPI=y

### Filesystems for optical media
- CONFIG_ISO9660_FS=y
- CONFIG_UDF_FS=y

## Rationale
Without an initrd, device-driver autoload from MODALIAS events does not occur.
Building these boot-path drivers into the kernel guarantees USB/NVMe/MMC/optical
and common HBA-backed root devices enumerate in time for root=PARTUUID=… to resolve.

# Following drivers were removed to match CORE kernel config
- IMA may interfere with EVE's TPM services so disable it
- DRM may prevent GPU pass-through. Disable it
- EVE never uses sound, disable it as well

rucoder committed:
[hwe] Disable IMA, DRM, and SOUND

- IMA may interfere with EVE's TPM services so disable it
- DRM may prevent GPU pass-through. Disable it
- EVE doesn't use sound cards on host side

Signed-off-by: Mikhail Malyshev <mike.malyshev@gmail.com>

eriknordmark reviewed (commented):
Seems like you also removed sound (CONFIG_SOUND, CONFIG_SND_*) or is that something different than the support for audio in/out?
We've seen use cases for vibration analysis using the sound support.

eriknordmark commented:
> * EVE never uses sound, disable it as well

What about an OCI container which wants to use sound? The shim/micro-VM kernel is the same as the EVE kernel so it will not be there any more.
(Note that I don't know if a container which uses sound, but we've had VM or containers use sound in the past.)

rucoder commented:
> > * EVE never uses sound, disable it as well
> 
> What about an OCI container which wants to use sound? The shim/micro-VM kernel is the same as the EVE kernel so it will not be there any more. (Note that I don't know if a container which uses sound, but we've had VM or containers use sound in the past.)

@eriknordmark good point, I'll double check what we have on regular kernel

rucoder commented:
> > * EVE never uses sound, disable it as well
> 
> What about an OCI container which wants to use sound? The shim/micro-VM kernel is the same as the EVE kernel so it will not be there any more. (Note that I don't know if a container which uses sound, but we've had VM or containers use sound in the past.)

@eriknordmark regular kernel has SOUND disabled

rucoder committed:
[hwe]: build-in boot-path drivers for no-initrd root (USB/NVMe/MMC/optical/RAID HBAs)

Booting without an initrd means the kernel cannot rely on early userspace to autoload modules
(udev/mdev/nlplug). Make storage and host drivers that can host the root filesystem built-in (=y)
so the kernel can discover and mount root directly across common platforms.

x86/PCI
-------
- CONFIG_VMD=y              # Intel VMD so NVMe behind VMD is visible early

Block / Virt
------------
- CONFIG_VIRTIO_BLK=y       # root on virtio disks without initrd

NVMe
----
- CONFIG_BLK_DEV_NVME=y     # NVMe root without module autoloading

SCSI / RAID HBAs (server boot devices)
--------------------------------------
- CONFIG_SCSI_HPSA=y
- CONFIG_MEGARAID_SAS=y
- CONFIG_SCSI_MPT3SAS=y     # common enterprise boot HBAs built-in

USB host & storage (USB stick / USB CD/DVD)
-------------------------------------------
- CONFIG_USB_XHCI_HCD=y
- CONFIG_USB_XHCI_PCI_RENESAS=y
- CONFIG_USB_XHCI_PLATFORM=y
- CONFIG_USB_EHCI_HCD=y
- CONFIG_USB_STORAGE=y
- CONFIG_USB_UAS=y          # UAS when available

MMC/SD (eMMC/SD cards)
----------------------
- CONFIG_MMC_SDHCI=y
- CONFIG_MMC_SDHCI_PCI=y
- CONFIG_MMC_SDHCI_ACPI=y
(Note: if root on MMC is required, ensure CONFIG_MMC_BLOCK=y as well.)

Filesystems for optical media
-----------------------------
- CONFIG_ISO9660_FS=y
- CONFIG_UDF_FS=y

Rationale
---------
Without an initrd, device-driver autoload from MODALIAS events does not occur.
Building these boot-path drivers into the kernel guarantees USB/NVMe/MMC/optical
and common HBA-backed root devices enumerate in time for root=PARTUUID=… to resolve.

Signed-off-by: Mikhail Malyshev <mike.malyshev@gmail.com>

eriknordmark reviewed (approved):
LGTM
